Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -113,12 +123,24 @@ class TIFFLoadingContext {
             return {};
         };
 
+        auto read_alpha = [&]() -> ErrorOr<u8> {
+            if (m_metadata.extra_samples().has_value()) {
+                auto const extra_sample = m_metadata.extra_samples().value();
+                if (extra_sample.size() > 0 && extra_sample[extra_sample.size() - 1] == ExtraSample::AssociatedAlpha) {
+                    return TRY(read_component(stream, bits_per_sample[3 + extra_sample.size() - 1]));
```

Comment:
Why `3`? What if there are additional unknown samples?

Does bits_per_sample contain bits both for main samples and extra_samples?